File: archived/v1/src/erlang/ch01/1.3/l0/src/parity.erl
-module(parity).
-export([is_even/1, is_odd/1, parity_bit/1]).

is_even(N) when is_integer(N) -> N rem 2 =:= 0.
is_odd(N)  when is_integer(N) -> N rem 2 =/= 0.

%% Return <<"even">> or <<"odd">> using the last bit
parity_bit(N) when is_integer(N) ->
    case (N band 1) of
        1 -> <<"odd">>;
        _ -> <<"even">>
    end.

File: archived/v1/src/c/ch01/1.3/l0/parity.c
#include "parity.h"

bool is_even(int64_t n) { return (n % 2) == 0; }
bool is_odd (int64_t n) { return (n % 2) != 0; }

const char *parity_bit(int64_t n) {
    return (n & 1) ? "odd" : "even";
}
